### PR TITLE
feat(ui): paginate the event log

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Paginate the event log and fetch additional events.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- View the React logs tab for a machine (change /summary to /logs).
- You should see < > arrows to the right of the search box.
- Use those to navigate forward and back through the events.
- As you navigate forward it'll load more events if they exist.

## Fixes

Fixes: canonical-web-and-design/maas-squad#2372.

### CODE REVIEW

**Review by anthonydillon — APPROVED** [review]
LGTM but could you switch to using the existing next and previous pattern in Vanilla.

File: ui/src/app/base/components/ArrowPagination/ArrowPagination.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  pageSize,
+  setCurrentPage,
+}: Props): JSX.Element => {
+  const onFirstPage = currentPage === 1;
+  const onLastPage =
+    itemCount === 0 || currentPage === Math.ceil(itemCount / pageSize);
+
+  return (
+    <>
+      <Button
```
[the remark this answers is not in the document]
Comment (anthonydillon) [reply ui/src/app/base/components/ArrowPagination/ArrowPagination.tsx:22]: Oh, I see the design is different again. Never mind, I'll file an issue on Vanilla.

File: ui/src/app/base/components/ArrowPagination/ArrowPagination.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        appearance="base"
+        className="u-no-margin--right u-no-margin--bottom"
+        disabled={onFirstPage}
+        onClick={() => {
```
Comment (anthonydillon) [inline ui/src/app/base/components/ArrowPagination/ArrowPagination.tsx:26]: Is it possible to add an `aria-label` to these buttons?

File: ui/src/app/base/components/ArrowPagination/ArrowPagination.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (huwshimi) [reply ui/src/app/base/components/ArrowPagination/ArrowPagination.tsx:22]: Yeah, I think this was to help users quickly scroll through events without scrolling down each time to find the next button.

File: ui/src/app/base/components/ArrowPagination/ArrowPagination.test.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        currentPage={1}
+        itemCount={50}
+        pageSize={25}
+        setCurrentPage={() => null}
+      />
+    );
+    expect(wrapper.find("Button").first().prop("disabled")).toBe(true);
+  });
+
+  it("disables the forward button when on the last page", () => {
```
[the remark this answers is not in the document]
Comment (huwshimi) [reply ui/src/app/base/components/ArrowPagination/ArrowPagination.test.tsx:31]: Will do.

File: ui/src/app/base/components/ArrowPagination/ArrowPagination.tsx (hunk 2 above)
Comment (huwshimi) [reply ui/src/app/base/components/ArrowPagination/ArrowPagination.tsx:26]: Yep, will do.

**Pull request merged**